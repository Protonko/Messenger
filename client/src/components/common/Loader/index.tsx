import type {FC} from 'react'
import classNames from 'classnames'
import {Sizes} from 'models/common/sizes'

export interface IPropsLoader {
  center?: boolean
  size?: Sizes
}

export const Loader: FC<IPropsLoader> = ({center, size = Sizes.MEDIUM}) => {
  const classNamesLoader = classNames(
    'loader',
    {'loader--center': center},
    {'loader--small': size === Sizes.SMALL},
    {'loader--large': size === Sizes.LARGE},
  )

  const renderLine = (_: string, index: number) => {
    return <div className="loader__line" key={index} />
  }

  return (
    <div className={classNamesLoader}>
      {new Array(12).fill('').map(renderLine)}
    </div>
  )
}
