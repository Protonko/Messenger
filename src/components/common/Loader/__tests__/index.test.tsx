import {shallow} from 'enzyme'
import {Loader} from 'components/common/Loader/index'

const ELEMENT_SELECTORS = {
  loader: 'loader',
}

const shallowCardComponent = () => (
  shallow(<Loader />)
)

describe('Loader', () => {
  it('Should render Loader component', () => {
    const component = shallowCardComponent()
    const wrapper = component.find(`.${ELEMENT_SELECTORS.loader}`)
    expect(wrapper.length).toBe(1)
  })

  it('Match snapshot', () => {
    const component = shallowCardComponent()

    expect(component).toMatchSnapshot()
  })
})
