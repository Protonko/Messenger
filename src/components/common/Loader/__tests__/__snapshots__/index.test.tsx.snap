// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Match snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="loader__line"
    key="0"
  />
  <div
    className="loader__line"
    key="1"
  />
  <div
    className="loader__line"
    key="2"
  />
  <div
    className="loader__line"
    key="3"
  />
  <div
    className="loader__line"
    key="4"
  />
  <div
    className="loader__line"
    key="5"
  />
  <div
    className="loader__line"
    key="6"
  />
  <div
    className="loader__line"
    key="7"
  />
  <div
    className="loader__line"
    key="8"
  />
  <div
    className="loader__line"
    key="9"
  />
  <div
    className="loader__line"
    key="10"
  />
  <div
    className="loader__line"
    key="11"
  />
</div>
`;
